tests/test_agents.py

```diff
@@ -1103,3 +1103,29 @@ async def test_continuation(self) -> None:
         # Check continuation of the search
         result = await tool.clinical_trials_search("Covid-19 vaccines", state)
         assert len(state.docs.docs) > trial_count, "Search was unable to continue"
+
+
+@pytest.mark.asyncio
+async def test_index_build_concurrency(agent_test_settings: Settings) -> None:
+
+    high_concurrency_settings = agent_test_settings.model_copy(deep=True)
+    high_concurrency_settings.agent.index.name = "high_concurrency"
+    high_concurrency_settings.agent.index.concurrency = 3
+    high_concurrency_settings.agent.index.batch_size = 3
```

Comment:
Do we really need `batch_size = 3` in this test? I think `concurrency = 3` alone should be good enough

Reply:
It seems so, but I don't really understand why.
I mean, isn't commiting every time the same as we did before?